proxy/src/control/destination/mod.rs

```diff
@@ -93,21 +95,7 @@ pub struct Metadata {
     dst_labels: Option<DstLabels>,
 
     /// How to verify TLS for the endpoint.
-    tls_identity: Option<Arc<TlsIdentity>>,
-}
-
-/// How to verify TLS for an endpoint.
-///
-/// XXX: This currently derives `PartialEq  and `Eq`, which is not entirely
-/// correct, as domain name equality ought to be case insensitive. However,
-/// `Metadata` must be `Eq`.
-#[derive(Debug, PartialEq, Eq)]
-pub enum TlsIdentity {
-    K8sPodNamespace {
-        controller_ns: String,
-        pod_ns: String,
-        pod_name: String,
-    },
+    tls_identity: Option<Arc<tls::Identity>>,
```

Comment:
TIOLI: Elsewhere (e.g. in `DstLabels`) we move the `Arc` inside of the publicly-exposed type,  do you thin it would make sense to do that here?